src/sentry/api/endpoints/system_options.py

```diff
@@ -72,10 +72,10 @@ def put(self, request):
                     options.delete(k)
                 else:
                     options.set(k, v)
-            except TypeError as e:
+            except (TypeError, AssertionError) as e:
                 return Response(
                     {
-                        "error": "invalid_type",
+                        "error": "invalid_type" if type(e) is TypeError else "invalid_option",
```

Comment:
Mind explaining why this was needed?

Reply:
Sure! I wanted to preserve the error string in the `TypeError` case, but figured it made sense to have a different error string in the `AssertionError` case. I went with `invalid_option` to keep the parallelism with `invalid_type` on the one hand and `unknown_option` (just above) on the other.

Reply:
Okay, I see that we also have an `unknown_option` error at the top which sounds similar. Should this be something like `immutable_option` or `unmodifiable_option` instead to clarify?

Reply:
Yeah, that can work. It looks like all five assertions beneath the relevant `OptionsManager` calls can be brought under the heading of immutability. Might be even better to have a custom exception for those vs. bare asserts, but there's a whole lot of asserting going on and I don't want to cut too deep.

https://github.com/getsentry/sentry/blob/848be85afc66acff6d430a6507cb5b8c59df96fc/src/sentry/options/manager.py#L77-L84

https://github.com/getsentry/sentry/blob/848be85afc66acff6d430a6507cb5b8c59df96fc/src/sentry/options/manager.py#L181-L184

Reply:
Done in e1d29ee0f4!

Reply:
Done in e1d29ee0f4!